code/network/multi_lua.cpp

```diff
@@ -0,0 +1,423 @@
+#include "multi_lua.h"
+
+#include "network/multimsgs.h"
+#include "network/multiutil.h"
+#include "scripting/ade.h"
+#include "scripting/ade_api.h"
+
+//This will send a byte more per userdata, but catch more faults due to players having different APIs
+//Also, it makes the remaining faults safer, since this way we'll only get invalid data, not accessing potentially invalid memory
+#define SAFE_MULTI_LUA_USERDATA_SIZES true
+
+static SCP_unordered_map<ushort, scripting::api::rpc_h_ref> rpc_map;
+
+enum class lua_net_data_type : uint8_t { NIL, BOOL, NUMBER, STRING8, STRING16, USERDATA, TABLE };
+
+union lua_packet_header {
+	struct {
+		unsigned short target : 13;
+		unsigned short isOrdered : 1;
+		unsigned short toServer : 1;
+		unsigned short toClient : 1;
+	} data;
+	unsigned short packed;
+};
+static_assert(sizeof(lua_packet_header) == 2, "Lua Packet Header is not properly packed!");
```

Comment:
As stated in our discord discussion, we can't guarantee that the variables are in the same order in different compilers, so we'll have to swtich to a packing function for the lua packet header.